rafthttp/http.go

```diff
@@ -72,6 +74,15 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if err := checkVersionCompability(serverVersion(r.Header), minClusterVersion(r.Header)); err != nil {
+		log.Printf("rafthttp: request ignored due to %v", err)
+		http.Error(w, errMsgIncompatibleVersion, http.StatusForbidden)
```

Comment:
This seems exactly what we do: The precondition given in one or more of the request-header fields evaluated to false when it was tested on the server.